### PR TITLE
feat(vowifi): retry discovery and report health when a configured line never resolves

### PR DESCRIPTION
**Pull request by selvakn**
## Summary

Fixes a real production incident: a configured `[[vowifi.line]] modem_port` line can be missed by `discover`'s single, one-shot startup scan (e.g. an EC20 modem that hadn't finished USB enumeration when `discover` ran), leaving it completely absent for the container's entire lifetime — not even reported as failed, invisible to `vowifi-status`, `healthcheck`, and every existing alert.

- **Detect** (Foundational): a configured `modem_port`/`modem_serial` override that matches no probed device is now recorded as `Rejection::NotFound`/`reason: "not_found"` in the resolution file, immediately, on every `discover` pass.
- **Retry & self-heal** (US1): when `discover`'s first pass resolves *zero* VoWiFi lines, a background thread re-runs `discover` every ~10s and starts the whole VoWiFi subsystem the moment any configured line resolves — no restart needed. Deliberately does **not** attempt to hot-add a line once a sibling line has already started (the shared `charon` daemon's P-CSCF plugin only reads its connection list once at its own process start; restarting it to pick up a late line would drop every other line's live calls) — documented as a known limitation in `spec.md`/`research.md`.
- **Visibility** (US2): `vowifi-status` now prints a `Configured line <id>: NOT RUNNING` section for any such line, and the container `healthcheck` reports unhealthy instead of the previous silent "healthy" — the exact blind spot in the original incident.
- **Alerting** (US3): a new `[alerts.line_discovery_failed]` Discord category fires once after ~3 minutes if a line still hasn't resolved, with a paired recovery notice if it later self-heals — reusing the existing `gsm_sip_bridge_critical_event_active` gauge rather than adding a new metric.

Full spec/plan/research/tasks under `specs/027-discover-retry-health/`.

## Test plan

- [x] `make format && make lint && make test` green after every commit (967 unit tests + all integration suites, zero failures)
- [x] New unit/integration tests per phase: `unmatched_overrides` (Foundational), `discover_retry_tick` (US1, 7 cases incl. the alerted→resolved edge case), `print_status`/`is_configured_line_failure` (US2), `healthcheck::evaluate` (US2, 3 new cases), `alerts::dispatch` end-to-end via wiremock incl. the Failure/Recovered pair and the `CRITICAL_EVENT_ACTIVE` gauge (US3), config parsing/docs (US3)
- [ ] **Not run**: `specs/027-discover-retry-health/quickstart.md`'s manual hardware verification — the only available modem/Docker environment is a live production container currently bridging real calls, and rebuilding/restarting it wasn't authorized for this session. Recommend running quickstart.md's steps deliberately before or after merge.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: gsm-sip-bridge/src/supervise/orchestrate.rs [hunk 1]
```diff
@@ -110,6 +121,414 @@ struct LineStartup<'a> {
...
+            }
+            let Ok(resolution) = crate::vowifi::discovery::read_line_resolution(&lines_file) else {
+                continue;
+            };
+            let still_missing: std::collections::HashSet<String> = resolution
+                .failed
+                .iter()
+                .filter(|f| f.reason == "not_found")
+                .map(|f| f.card_id.clone())
+                .collect();
```
[the remark this answers is not in the document]
Comment (selvakn) [reply gsm-sip-bridge/src/supervise/orchestrate.rs]: Fixed in 64e0039: `still_missing` is no longer derived from `resolution.failed` filtered to `reason=="not_found"`. It's now computed by `still_missing_from_resolution`, which checks membership in `resolution.lines` directly — so a pending line that changes rejection reason on a later tick (e.g. `not_found` → `sim_unreadable` once the modem is actually detected but its SIM isn't) still correctly counts as missing, regardless of what `resolution.failed`'s reason says. Added two direct unit tests for this (`still_missing_from_resolution_stays_missing_when_the_failure_reason_changes`, `..._clears_once_the_id_is_an_actual_resolved_line`) plus verified live against real EC20 hardware.

File: gsm-sip-bridge/src/vowifi/mod.rs [hunk 2]
```diff
@@ -1899,6 +1926,21 @@ pub fn print_status(config: &VowifiConfig) -> ExitCode {
...
+    // override that failed to become a line, printed after the resolved
+    // ones — contracts/vowifi-status-output.md.
+    for failed in resolution
+        .failed
+        .iter()
+        .filter(|f| is_configured_line_failure(f))
+    {
+        println!(
+            "Configured line {} (from config.toml): NOT RUNNING",
+            failed.card_id
```
[the remark this answers is not in the document]
Comment (selvakn) [reply gsm-sip-bridge/src/vowifi/mod.rs:1949]: Fixed in 64e0039: added real provenance tracking instead of the reason-string exclusion. `FailedLine` now carries a `configured: bool`, set at the point `resolve_lines`/`unmatched_overrides` actually know whether a candidate matched an explicit `[[vowifi.line]]` override (checked via `is_overridden_to_vowifi` even in the classify-failure branch, before the pinned/unpinned partition happens). `is_configured_line_failure` now checks `failed.configured` directly, so an auto-discovered modem's `sim_unreadable`/`sim_locked`/`no_at_port` is never mislabeled as a configured line from config.toml — and as a bonus this also correctly now flags the case where an operator pins *more* lines than `max_lines` allows (previously excluded entirely by the reason-string check, even though that genuinely is a configured line not running).

File: gsm-sip-bridge/src/supervise/orchestrate.rs [hunk 3]
```diff
@@ -110,6 +121,433 @@ struct LineStartup<'a> {
...
+/// set of lines *did* resolve, permanently dropped the still-broken id from
+/// `pending` so it was never retried again, and could fire a false
+/// `Recovered` alert for a line that was never actually started.
+fn still_missing_from_resolution(
+    pending: &std::collections::HashMap<String, std::time::Instant>,
+    resolution: &crate::vowifi::discovery::LineResolution,
+) -> std::collections::HashSet<String> {
+    pending
+        .keys()
+        .filter(|id| !resolution.lines.iter().any(|l| &l.card_id == *id))
```
[the remark this answers is not in the document]
Comment (selvakn) [reply gsm-sip-bridge/src/supervise/orchestrate.rs]: Good catch — fixed in 89a4284. `pending`'s keys are always the configured override's own `modem_port`/`modem_serial` string (from `unmatched_overrides`), never a USB-derived `card_id`, so comparing them straight against `resolution.lines`' `card_id` could never succeed once a line actually resolved. Now matched the same way `resolve_lines`/`unmatched_overrides` establish the pinning in the first place: a `modem_port` key against `LineResolutionEntry.modem_port` verbatim, a `modem_serial` key re-derived via `derive_module_id` and compared against `card_id` (mirrors how `ProbedModem.card_id` itself gets derived from the raw USB serial in `scan_all_inner`). Added three tests: port-based resolution, serial-based resolution, and an explicit regression test asserting the pending key and `card_id` differ yet still resolve correctly.

File: gsm-sip-bridge/src/supervise/orchestrate.rs [hunk 4]
```diff
@@ -110,6 +121,455 @@ struct LineStartup<'a> {
...
+fn still_missing_from_resolution(
+    pending: &std::collections::HashMap<String, std::time::Instant>,
+    resolution: &crate::vowifi::discovery::LineResolution,
+) -> std::collections::HashSet<String> {
+    pending
+        .keys()
+        .filter(|id| {
+            !resolution.lines.iter().any(|l| {
+                l.modem_port == **id || crate::modules::discovery::derive_module_id(id) == l.card_id
+            })
```
[the remark this answers is not in the document]
Comment (selvakn) [reply gsm-sip-bridge/src/supervise/orchestrate.rs]: Fixed in c20c27a — and this time by removing the inference rather than refining it.

You are right that `derive_module_id` is lossy (last six alphanumerics, uppercased), so `AAAAAAAAAAabcdef` and `BBBBBBBBBBABCDEF` collapse to the same `card_id`. Two P1s in a row on this function both came from trying to reconstruct the override→line correspondence *after the fact*, so I stopped doing that: `resolve_lines` already holds the exact matched override (`override_for`), so the resolution now records it.

- `ResolvedLine`/`LineResolutionEntry` carry `configured_identifier`.
- `unmatched_overrides` (missing side) and `resolve_one_line` (resolved side) both derive it from one shared `override_identifier()`, so they cannot drift.
- `still_missing_from_resolution` is now an exact set-membership test — no `card_id`, no `derive_module_id`, nothing reconstructed.
- `#[serde(default)]` so an older resolution file left in `/tmp` across a `docker restart` reads back as `None` (= "not a configured line"), which fails safe: the loop keeps waiting rather than falsely reporting recovery.

Tests added: a round-trip asserting the identifier is byte-identical across the missing→resolved transition, your exact collision scenario (colliding suffixes, differing case), and an auto-discovered line never satisfying a pending override.

Reviewing that path also surfaced a regression from my previous commit, which I have fixed here: `probe_sim_status_at` is shared by the one-shot `discover` scan **and** `scan_modules` ongoing rescans, so the `AT+CFUN` SIM recovery I added would have power-cycled a modem radio during routine rescans — including a circuit-switched modem mid-call, which `skip_card_ids` does not cover (it only shields VoWiFi lines). It is now an explicit `SimRecovery` opt-in that only `discover` passes, with a guard test that fails if any in-module scan call site hard-codes it.

Verified on real EC20 hardware: the resolved line reports `configured_identifier: "/dev/ttyUSB3"`, matching the pending key exactly.
